lstchain/irf/hdu_table.py

```diff
@@ -0,0 +1,424 @@
+
+    hdu_index = fits.BinTableHDU(
+        hdu_index_table, header=hdu_index_header, name="HDU INDEX"
+    )
+    hdu_index_list = fits.HDUList([fits.PrimaryHDU(), hdu_index])
+
+    return hdu_index_list
+
+
+def create_event_list(
+    data, run_number, source_name, source_pos, effective_time, elapsed_time
+):
+    """
+    Create the event_list BinTableHDUs from the given data
+
+    Parameters
+    ----------
+        Data: DL2 data file
+                'astropy.table.QTable'
+        Run: Run number
+                Int
+        Source_name: Name of the source
+                Str
+        Source_pos: Ra/Dec position of the source
+                'astropy.coordinates.SkyCoord'
+        Effective_time: Effective time of triggered events of the run
+                Float
+        Elapsed_time: Total elapsed time of triggered events of the run
+                Float
+    Returns
+    -------
+        Events HDU:  `astropy.io.fits.BinTableHDU`
+        GTI HDU:  `astropy.io.fits.BinTableHDU`
+        Pointing HDU:  `astropy.io.fits.BinTableHDU`
+    """
+
+    tel_list = np.unique(data["tel_id"])
+
+    # Timing parameters
+    t_start = data["dragon_time"].value[0]
+    t_stop = data["dragon_time"].value[-1]
+    time = Time(data["dragon_time"], format="unix", scale="utc")
+    date_obs = time[0].to_value("iso", "date")
+    mean_time = Time(
+        data["dragon_time"].value.mean()*u.s, format="unix", scale="utc"
+    )
+
+    # Position parameters
+    reco_alt = data["reco_alt"]
+    reco_az = data["reco_az"]
+    pointing_alt = data["pointing_alt"]
+    pointing_az = data["pointing_az"]
+
+    src_sky_pos = SkyCoord(
+        alt=reco_alt, az=reco_az, frame=AltAz(obstime=time, location=location)
+    ).transform_to(frame="icrs")
+    tel_pnt_sky_pos = SkyCoord(
+        alt=pointing_alt.mean(),
+        az=pointing_az.mean(),
+        frame=AltAz(obstime=mean_time, location=location),
+    ).transform_to(frame="icrs")
+
+    # Observation modes
+    source_pointing_diff = source_pos.separation(tel_pnt_sky_pos).to_value()
+
+    if round(source_pointing_diff, 1) == wobble_offset:
+        mode = "WOBBLE"
+    elif round(source_pointing_diff, 1) > 1:
+        mode = "OFF"
+    elif round(source_pointing_diff, 1) == 0.0:
+        mode = "ON"
+    else:
+        # Nomenclature is to be worked out or have a separate way to mark mispointings
+        mode = "UNDETERMINED"
+
+    log.info(
+        "Source pointing difference with camera pointing"
+        f" is {source_pointing_diff:.3f} deg"
+    )
+
+    event_table = QTable(
+        {
+            "EVENT_ID": data["event_id"],
+            "TIME": data["dragon_time"],
+            "RA": src_sky_pos.ra.to(u.deg),
+            "DEC": src_sky_pos.dec.to(u.deg),
+            "ENERGY": data["reco_energy"],
+        }
+    )
+    gti_table = QTable(
+        {"START": u.Quantity(t_start, ndmin=1), "STOP": u.Quantity(t_stop, ndmin=1)}
+    )
+    pnt_table = QTable()
```

Comment:
Do I see it correctly, that the pointing table is empty? Its never filed, right?

Reply:
To be a valid table, it needs these mandatory columns:

https://gamma-astro-data-formats.readthedocs.io/en/latest/events/pointing.html#mandatory-columns

Reply:
Yes, I corrected this.